widget/embedded/src/utils/swap.ts

```diff
@@ -304,6 +305,22 @@ export function getSwapButtonState(params: {
     };
   }
   const requiredChains = getQuoteChains({ quote, filter: 'required' });
+
+  if (requiredChains.length === 1) {
+    const walletAddressError =
+      !selectedWallets.customDestination &&
+      selectedWallets.sourceWallet?.address !==
+        selectedWallets.destinationWallet?.address;
+
+    if (walletAddressError) {
+      return {
+        title: swapButtonTitles().swap,
+        action: 'show-wallet-address-error',
+        disabled: false,
+      };
+    }
+  }
```

Comment:
Why do we need nested ifs? would this also work?
```suggestion
if (
  requiredChains.length === 1 &&
  !selectedWallets.customDestination &&
  selectedWallets.sourceWallet?.address !== selectedWallets.destinationWallet?.address
) {
  return {
    title: swapButtonTitles().swap,
    action: 'show-wallet-address-error',
    disabled: false,
  };
}
```
Also please add a comment explaining what are you checking in here.

Reply:
I don’t agree with you. This error occurs only when executing a simple on-chain swap, so we should check for it only in that specific case. I also think extracting the long condition into a variable improves readability — but I’ve added a comment for clarity as well.